Prepare MomentumEventDispatcher to be used with UI-side compositing

smfr opened the pull request:
#### dacbd7c36056647c1867f34236ade6815002e1f1
<pre>
Prepare MomentumEventDispatcher to be used with UI-side compositing
<a href="https://bugs.webkit.org/show_bug.cgi?id=252537">https://bugs.webkit.org/show_bug.cgi?id=252537</a>
rdar://105635572

Reviewed by Tim Horton.

Give MomentumEventDispatcher a Client interface that&apos;s implemented by EventDispatcher.

MomentumEventDispatcher needs public inheritance from IPC::MessageReceiver so we can make a WeakPtr
from it.

* Source/WebKit/WebProcess/WebPage/EventDispatcher.cpp:
(WebKit::EventDispatcher::setScrollingAccelerationCurve):
(WebKit::EventDispatcher::handleSyntheticWheelEvent):
(WebKit::EventDispatcher::flushMomentumEventLoggingSoon):
* Source/WebKit/WebProcess/WebPage/EventDispatcher.h:
* Source/WebKit/WebProcess/WebPage/MomentumEventDispatcher.cpp:
(WebKit::MomentumEventDispatcher::MomentumEventDispatcher):
(WebKit::MomentumEventDispatcher::dispatchSyntheticMomentumEvent):
(WebKit::MomentumEventDispatcher::didEndMomentumPhase):
* Source/WebKit/WebProcess/WebPage/MomentumEventDispatcher.h:

Canonical link: <a href="https://commits.webkit.org/260522@main">https://commits.webkit.org/260522@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e343e067e8417816797435421d3bb31c0e0b7164

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108538 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17635 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41388 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117644 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/117842 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19086 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8913 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100766 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114305 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14303 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97530 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42271 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96280 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29171 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83974 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10445 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30520 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11202 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7430 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/16592 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50115 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7269 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12790 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/WebProcess/WebPage/EventDispatcher.cpp:
```diff
@@ -342,4 +335,26 @@ void EventDispatcher::pageScreenDidChange(PageIdentifier pageID, PlatformDisplay
+}
+
+#if ENABLE(MOMENTUM_EVENT_DISPATCHER_TEMPORARY_LOGGING)
+void EventDispatcher::flushMomentumEventLoggingSoon()
```
Why does this have to be delayed?

smfr replied:
So that it didn't interfere with performance during the gesture. Note that this logging is off by default.

smfr committed:
Prepare MomentumEventDispatcher to be used with UI-side compositing
https://bugs.webkit.org/show_bug.cgi?id=252537
rdar://105635572

Reviewed by Tim Horton.

Give MomentumEventDispatcher a Client interface that's implemented by EventDispatcher.

MomentumEventDispatcher needs public inheritance from IPC::MessageReceiver so we can make a WeakPtr
from it.

* Source/WebKit/WebProcess/WebPage/EventDispatcher.cpp:
(WebKit::EventDispatcher::setScrollingAccelerationCurve):
(WebKit::EventDispatcher::handleSyntheticWheelEvent):
(WebKit::EventDispatcher::flushMomentumEventLoggingSoon):
* Source/WebKit/WebProcess/WebPage/EventDispatcher.h:
* Source/WebKit/WebProcess/WebPage/MomentumEventDispatcher.cpp:
(WebKit::MomentumEventDispatcher::MomentumEventDispatcher):
(WebKit::MomentumEventDispatcher::dispatchSyntheticMomentumEvent):
(WebKit::MomentumEventDispatcher::didEndMomentumPhase):
* Source/WebKit/WebProcess/WebPage/MomentumEventDispatcher.h:

Canonical link: https://commits.webkit.org/260522@main
